app/hooks/use-latest-nonce.ts

```diff
@@ -0,0 +1,30 @@
+import { useCallback } from 'react';
+import useSWR from 'swr';
+import { useSelector } from 'react-redux';
+
+import { Api } from '@api/api';
+import { RootState } from '@store/index';
+import { selectAddress } from '@store/keys';
+import { selectActiveNodeApi } from '@store/stacks-node';
+import { useMempool } from '@hooks/use-mempool';
+import BigNumber from 'bignumber.js';
+
+export function useLatestNonce() {
+  const { address, activeNode } = useSelector((state: RootState) => ({
+    address: selectAddress(state),
+    activeNode: selectActiveNodeApi(state),
+  }));
+  const { outboundMempoolTxs } = useMempool();
+
+  const nonceFetcher = useCallback(() => new Api(activeNode.url).getNonce(address || ''), [
+    activeNode.url,
+    address,
+  ]);
+  const { data } = useSWR('nonce', nonceFetcher);
+
+  if (!data) return { nonce: 0 };
+
+  return {
+    nonce: new BigNumber(data.nonce).plus(outboundMempoolTxs.length).toNumber(),
```

Comment:
Seems like this should be `nonce = max(account.nonce, max(mempool-nonces))` ?

Reply:
Yes.. now that I think about it this might not be great as-is. I'd suggest `max(account.nonce, max(mempoolNonces) + 1)`. But I'd wonder if you could still end up with issues if there is a "gap" in nonces. If the mempool tx's weren't rejected, you'd keep incrementing, making it worse.

I do like your approach here because it's stateless, but thinking about this, I'm happy with my method of storing the `{lastTxSentNonce, lastTxSentBlockHeight}`